src/strategies/mixins/FluentConstantExtractionMixin.ts

```diff
@@ -36,11 +36,18 @@ export function ConstantExtractionBased<TBase extends MixinStrategy>(Base: TBase
       })
 
       const numericsAndPunctuators = filteredTokens.reduce(function (acc, token, index) {
-        if (token.type === 'Numeric')
+        if (token.type === 'Numeric') {
+          const leftPunctuatorIndex = (filteredTokens[index - 1] !== undefined &&
+            filteredTokens[index - 1].value !== '-') ? 1 : 2
```

Comment:
I'm starting to feel that a token-based parser would not scale very well...

Reply:
I understand your point but apart from `numeric` we are only interested in `string` tokens. So I think we are good.

Reply:
Btw this code section you used for making your statement has only two types of tokens: `punctuator` or `numeric`